### PR TITLE
[RLlib] Remove `return_info` from `reset()` in `pettingzoo_env.py`.

### LINKED ISSUE #32889: [RLlib] ParallelPettingZooEnv TypeError: reset() got an unexpected keyword argument 'return_info'
### What happened + What you expected to happen

I opened an issue with PettingZoo which I believe is the main cause of the problem, but it also may be related to RLlib, posting this here in case someone who's worked on the ParallelPettingZooEnv class could help diagnose the problem: https://github.com/Farama-Foundation/PettingZoo/issues/889

### Versions / Dependencies

gym==0.23.1
Gymnasium==0.26.3
numpy==1.23.5
PettingZoo==1.22.3
Pillow==9.4.0
pygame==2.1.2
ray==2.3.0
SuperSuit==3.7.1
tianshou==0.4.11
torch==1.13.1

### Reproduction script

Basic working example not using ray (to show that the env works on its own, the pre-processing steps with ss aren't the problem afaik):

```
import supersuit as ss

from pettingzoo.butterfly import pistonball_v6

def env_creator(args):
    env = pistonball_v6.parallel_env(
        n_pistons=20,
        time_penalty=-0.1,
        continuous=True,
        random_drop=True,
        random_rotate=True,
        ball_mass=0.75,
        ball_friction=0.3,
        ball_elasticity=1.5,
        max_cycles=125,
        render_mode="human"
    )
    env = ss.color_reduction_v0(env, mode="B")
    env = ss.dtype_v0(env, "float32")
    env = ss.resize_v1(env, x_size=84, y_size=84)
    env = ss.normalize_obs_v0(env, env_min=0, env_max=1)
    env = ss.frame_stack_v1(env, 3)
    return env

if __name__ == "__main__":
    env = env_creator({})
    env.reset()

    while env.agents:
        actions = {agent: env.action_space(agent).sample() for agent in env.agents}
        observations, rewards, terminations, truncations, infos = env.step(actions)

```

rllib_pistonball.py:
```
"""Uses Ray's RLLib to train agents to play Pistonball.

Author: Rohan (https://github.com/Rohan138)
"""

import os

import ray
import supersuit as ss
from ray import tune
from ray.rllib.algorithms.ppo import PPOConfig
from ray.rllib.env.wrappers.pettingzoo_env import ParallelPettingZooEnv
from ray.rllib.models import ModelCatalog
from ray.rllib.models.torch.torch_modelv2 import TorchModelV2
from ray.tune.registry import register_env
from torch import nn

from pettingzoo.butterfly import pistonball_v6

# raise NotImplementedError(
#     "There are currently bugs in this tutorial, we will fix them soon."
# )


class CNNModelV2(TorchModelV2, nn.Module):
    def __init__(self, obs_space, act_space, num_outputs, *args, **kwargs):
        TorchModelV2.__init__(self, obs_space, act_space, num_outputs, *args, **kwargs)
        nn.Module.__init__(self)
        self.model = nn.Sequential(
            nn.Conv2d(3, 32, [8, 8], stride=(4, 4)),
            nn.ReLU(),
            nn.Conv2d(32, 64, [4, 4], stride=(2, 2)),
            nn.ReLU(),
            nn.Conv2d(64, 64, [3, 3], stride=(1, 1)),
            nn.ReLU(),
            nn.Flatten(),
            (nn.Linear(3136, 512)),
            nn.ReLU(),
        )
        self.policy_fn = nn.Linear(512, num_outputs)
        self.value_fn = nn.Linear(512, 1)

    def forward(self, input_dict, state, seq_lens):
        model_out = self.model(input_dict["obs"].permute(0, 3, 1, 2))
        self._value_out = self.value_fn(model_out)
        return self.policy_fn(model_out), state

    def value_function(self):
        return self._value_out.flatten()


def env_creator(args):
    env = pistonball_v6.parallel_env(
        n_pistons=20,
        time_penalty=-0.1,
        continuous=True,
        random_drop=True,
        random_rotate=True,
        ball_mass=0.75,
        ball_friction=0.3,
        ball_elasticity=1.5,
        max_cycles=125,
    )
    env = ss.color_reduction_v0(env, mode="B")
    env = ss.dtype_v0(env, "float32")
    env = ss.resize_v1(env, x_size=84, y_size=84)
    env = ss.normalize_obs_v0(env, env_min=0, env_max=1)
    env = ss.frame_stack_v1(env, 3)
    return env


if __name__ == "__main__":
    ray.init(local_mode=True)

    env_name = "pistonball_v6"

    register_env(env_name, lambda config: ParallelPettingZooEnv(env_creator(config)))
    ModelCatalog.register_custom_model("CNNModelV2", CNNModelV2)

    config = (
        PPOConfig()
        .rollouts(num_rollout_workers=4, rollout_fragment_length=128)
        .training(
            train_batch_size=512,
            lr=2e-5,
            gamma=0.99,
            lambda_=0.9,
            use_gae=True,
            clip_param=0.4,
            grad_clip=None,
            entropy_coeff=0.1,
            vf_loss_coeff=0.25,
            sgd_minibatch_size=64,
            num_sgd_iter=10,
        )
        .environment(env=env_name, clip_actions=True)
        .debugging(log_level="ERROR")
        .framework(framework="torch")
        .resources(num_gpus=int(os.environ.get("RLLIB_NUM_GPUS", "0")))
    )

    tune.run(
        "PPO",
        name="PPO",
        stop={"timesteps_total": 5000000},
        checkpoint_freq=10,
        local_dir="~/ray_results/" + env_name,
        config=config.to_dict(),
    )
```

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by elliottower**
## Why are these changes needed?

The `return_info` reset() parameter has been removed from the PettingZoo API for a few weeks: https://github.com/Farama-Foundation/PettingZoo/pull/890

Render() also has not taken the  render_mode argument since October of 2022: https://github.com/Farama-Foundation/PettingZoo/commit/a74a933237a73de84239b088dd16c3067a55e441

Motivation is that current pettingzoo version cannot run with RLlib without modifying these files locally to get rid of return_info (have tested with both parallel and aec envs, see https://github.com/Farama-Foundation/PettingZoo/pull/899/)

## Related issue number

Closes #32889 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by elliottower** [comment]
The DCO check failed due to commits being not signed, but I have 
`git config --global commit.gpgsign true` and it says all of my commits are verified. Followed the rebase instructions from the error message. Now I see there are some other error messages but I can't figure out what the problem could be. Can someone help figure these out? 

My only thought is that maybe the internal tests are using an older version of gymnasium/pettingzoo which still has return_info. Edit: I found this file for rllib with version 1.22.1, whereas 1.22.3 is the most widely used and 1.22.4 is the current release as of today.

**Comment by elliottower** [comment]
@ArturNiederfahrenhorst any chance you could help look this over? I would like to use RLlib with the most recent PettingZoo release but this makes it impossible without locally modifying it

**Review by sven1977 — APPROVED** [review]
Thanks for the fix @elliottower !

File: rllib/env/wrappers/pettingzoo_env.py [hunk 1]
```diff
@@ -154,7 +154,7 @@ def close(self):
         self.env.close()
 
     def render(self):
-        return self.env.render(self.render_mode)
+        return self.env.render()
```
Comment (avnishn) [inline rllib/env/wrappers/pettingzoo_env.py:157]: in the newest petting zoo version, are render modes no longer configurable? 

if so the whole pr looks p much looks good to me

**Review by avnishn — COMMENTED** [review]
This looks good to me. Can you pull ray/master into your pr that way we can see the latest tests run?

**Comment by avnishn** [comment]
oh and please run the linter on your changes ./ci/lint/format.sh

**Review by sven1977 — CHANGES_REQUESTED** [review]
Awesome PR. Still some minor fixes to do before we can merge this. See my comments.
Thanks again! :)

File: python/requirements/ml/requirements_rllib.txt [hunk 2]
```diff
@@ -15,7 +15,7 @@ kaggle_environments==1.7.11
 #mlagents==0.28.0
 mlagents_envs==0.28.0
 # For tests on PettingZoo's multi-agent envs.
-pettingzoo==1.22.1; python_version >= '3.7'
+pettingzoo==1.22.4; python_version >= '3.7'
```
[the remark this answers is not in the document]
Comment (avnishn) [reply python/requirements/ml/requirements_rllib.txt:18]: we should actually be able to bump the gym version with no problems here. Can you please do so @elliottower. I'd be happy to assist you in the upgrade if any issues come up.

[the remark this answers is not in the document]
Comment (elliottower) [reply python/requirements/ml/requirements_rllib.txt:18]: Okay I'll change it to the most recent versions of all of them. Gymnasium had a release 28 late last week and 28.1 a day or two later with hotfixes. The update made a slight change to the Spaces, such as Sequence action space, which led to an error in the PettingZoo CI, which I've now fixed (previous spaces.Sequence() calls need to specify spaces.Sequence(stack=True) to get the same behavior as before). I've fixed that issue and the PettingZoo CI tests there are now passing. I did a search and found no uses of spaces.Sequence in Ray's repo so I think that should be fine, but I can do the pytests and try to find other potential issues with gymnasium 28.1. 

We are finishing up a PettingZoo release to fix this gym 28.1, which we are hoping to get out by the end of this week. I can run the tests with the master branch of PettingZoo to ensure they pass and start debugging errors with them now, and then once the full release is out then I can change the requirements to list the new version of PettingZoo.

**Comment by elliottower** [comment]
> Awesome PR. Still some minor fixes to do before we can merge this. See my comments. Thanks again! :)

Could you help me diagnose the problems with the tests which are failing? It says 100 successful and 11 failing. 

I'm running the pytest for pettingzoo locally to try and see if I can find any issue but can't see anything wrong, not sure why my change would affect things like ci-build-pr/java

**Comment by elliottower** [comment]
Also I'm trying to install the repository locally and having issues, I ran `pip install -e . ` in the `/python/` directory, and it seemed to work at first but then had this error later 
```
 Analyzing: 2 targets (49 packages loaded, 1520 targets configured)
    ERROR: /private/var/tmp/_bazel_elliottower/fd84858c11d357b636f50f21ff877190/external/bazel_tools/platforms/BUILD:89:6: in alias rule @bazel_tools//platforms:windows: Constraints from @bazel_tools//platforms have been removed. Please use constraints from @platforms repository embedded in Bazel, or preferably declare dependency on https://github.com/bazelbuild/platforms. See https://github.com/bazelbuild/bazel/issues/8622 for details.
    ERROR: /private/var/tmp/_bazel_elliottower/fd84858c11d357b636f50f21ff877190/external/bazel_tools/platforms/BUILD:89:6: Analysis of target '@bazel_tools//platforms:windows' failed
    ERROR: /Users/elliottower/Documents/GitHub/ray/cpp/BUILD.bazel:6:10: While resolving toolchains for target //cpp:libray_api.so: invalid registered toolchain '@bazel_skylib//toolchains/unittest:cmd_toolchain':
    ERROR: Analysis of target '//cpp:ray_cpp_pkg' failed; build aborted:
    INFO: Elapsed time: 211.165s
    INFO: 0 processes.
    FAILED: Build did NOT complete successfully (55 packages loaded, 1741 targets configured)
    /Users/elliottower/anaconda3/envs/ray/lib/python3.8/site-packages/setuptools/command/easy_install.py:144: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
      warnings.warn(
    /Users/elliottower/anaconda3/envs/ray/lib/python3.8/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
      warnings.warn(
    Traceback (most recent call last):
      File "<string>", line 2, in <module>
      File "<pip-setuptools-caller>", line 34, in <module>
      File "/Users/elliottower/Documents/GitHub/ray/python/setup.py", line 768, in <module>
        setuptools.setup(
      File "/Users/elliottower/anaconda3/envs/ray/lib/python3.8/site-packages/setuptools/__init__.py", line 87, in setup
        return distutils.core.setup(**attrs)
      File "/Users/elliottower/anaconda3/envs/ray/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 185, in setup
        return run_commands(dist)
      File "/Users/elliottower/anaconda3/envs/ray/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
        dist.run_commands()
      File "/Users/elliottower/anaconda3/envs/ray/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
        self.run_command(cmd)
      File "/Users/elliottower/anaconda3/envs/ray/lib/python3.8/site-packages/setuptools/dist.py", line 1208, in run_command
        super().run_command(command)
      File "/Users/elliottower/anaconda3/envs/ray/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
        cmd_obj.run()
      File "/Users/elliottower/anaconda3/envs/ray/lib/python3.8/site-packages/setuptools/command/develop.py", line 34, in run
        self.install_for_development()
      File "/Users/elliottower/anaconda3/envs/ray/lib/python3.8/site-packages/setuptools/command/develop.py", line 114, in install_for_development
        self.run_command('build_ext')
      File "/Users/elliottower/anaconda3/envs/ray/lib/python3.8/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
        self.distribution.run_command(command)
      File "/Users/elliottower/anaconda3/envs/ray/lib/python3.8/site-packages/setuptools/dist.py", line 1208, in run_command
        super().run_command(command)
      File "/Users/elliottower/anaconda3/envs/ray/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
        cmd_obj.run()
      File "/Users/elliottower/Documents/GitHub/ray/python/setup.py", line 756, in run
        return pip_run(self)
      File "/Users/elliottower/Documents/GitHub/ray/python/setup.py", line 657, in pip_run
        build(True, BUILD_JAVA, True)
      File "/Users/elliottower/Documents/GitHub/ray/python/setup.py", line 605, in build
        return bazel_invoke(
      File "/Users/elliottower/Documents/GitHub/ray/python/setup.py", line 372, in bazel_invoke
        result = invoker([cmd] + cmdline, *args, **kwargs)
      File "/Users/elliottower/anaconda3/envs/ray/lib/python3.8/subprocess.py", line 364, in check_call
        raise CalledProcessError(retcode, cmd)
    subprocess.CalledProcessError: Command '['bazel', 'build', '--verbose_failures', '--', '//:ray_pkg', '//cpp:ray_cpp_pkg']' returned non-zero exit status 1.
    [end of output]

```

**Comment by elliottower** [comment]
@sven1977 @avnishn any ideas about how to fix the failing tests? Re ran them and it’s the same ones failing but they don’t appear to be related to my changes as far as I can see at least

**Comment by sven1977** [comment]
I think the only two tests that are of relevance are those that are failing with this error here:
```
Traceback (most recent call last):

ImportError:
 cannot import name 'BaseParallelWraper' from 
'pettingzoo.utils.wrappers' 
(/opt/miniconda/lib/python3.7/site-packages/pettingzoo/utils/wrappers/__init__.py)
```
It does seem like a supersuit/pettingzoo mismatch issue. ??

**Comment by elliottower** [comment]
> I think the only two tests that are of relevance are those that are failing with this error here:
> 
> ```
> Traceback (most recent call last):
> 
> ImportError:
>  cannot import name 'BaseParallelWraper' from 
> 'pettingzoo.utils.wrappers' 
> (/opt/miniconda/lib/python3.7/site-packages/pettingzoo/utils/wrappers/__init__.py)
> ```
> 
> It does seem like a supersuit/pettingzoo mismatch issue. ??

Ah yeah this was a typo which was fixed in the most recent pettingzoo update, thanks for pointing it out. I’ll work on getting those issues fixed. Going to wait to finalize this until we release the next major release though.

**Comment by elliottower** [comment]
Going to do a new PR because this one seems to have gotten messed up from a re-signing issue

**Pull request closed without merging**